### ISSUE TITLE
Evaluate a fixed validation dataset while training using fit_generator

### ISSUE THREAD

**Issue by vikiQiu** [keras-team/keras#13173]
My keras version is 2.2.4
I want to evaluate on all the validation data after each training epoch. While in my experiments, even the learning rate is set to 0 and only 1 step for training, my validation loss is changed greatly.
My parameters:
```
--GPU_COUNT=1
--IMAGES_PER_GPU=1 
--STEPS_PER_EPOCH=1 
--VALIDATION_STEPS=253 
--LEARNING_RATE=0
```

Results are:
```
Epoch 9/20
1/1 [==============================] - 43s 43s/step - loss: 1.9891 - rpn_class_loss: 3.1015e-05 - rpn_bbox_loss: 0.0531 - mrcnn_class_loss: 1.2486 - mrcnn_bbox_loss: 0.3533 - mrcnn_mask_loss: 0.3339 - val_loss: 2.4936 - val_rpn_class_loss: 0.0338 - val_rpn_bbox_loss: 1.2758 - val_mrcnn_class_loss: 0.4691 - val_mrcnn_bbox_loss: 0.3081 - val_mrcnn_mask_loss: 0.4068
Copying local_tb_dir(/tmp/model_dir/hx_cocoh5_072906) to tb_logdir(../jarvis_hx_output/tensorboard_0729/hx_cocoh5_072906) ...
Epoch 10/20
1/1 [==============================] - 42s 42s/step - loss: 1.9890 - rpn_class_loss: 3.1015e-05 - rpn_bbox_loss: 0.0531 - mrcnn_class_loss: 1.2485 - mrcnn_bbox_loss: 0.3533 - mrcnn_mask_loss: 0.3339 - val_loss: 2.5947 - val_rpn_class_loss: 0.1059 - val_rpn_bbox_loss: 0.8440 - val_mrcnn_class_loss: 0.6088 - val_mrcnn_bbox_loss: 0.5727 - val_mrcnn_mask_loss: 0.4632
Copying local_tb_dir(/tmp/model_dir/hx_cocoh5_072906) to tb_logdir(../jarvis_hx_output/tensorboard_0729/hx_cocoh5_072906) ...
Epoch 11/20
1/1 [==============================] - 43s 43s/step - loss: 2.0009 - rpn_class_loss: 3.1015e-05 - rpn_bbox_loss: 0.0531 - mrcnn_class_loss: 1.2605 - mrcnn_bbox_loss: 0.3533 - mrcnn_mask_loss: 0.3339 - val_loss: 3.2869 - val_rpn_class_loss: 0.0037 - val_rpn_bbox_loss: 1.0370 - val_mrcnn_class_loss: 0.9170 - val_mrcnn_bbox_loss: 0.4865 - val_mrcnn_mask_loss: 0.8428
Copying local_tb_dir(/tmp/model_dir/hx_cocoh5_072906) to tb_logdir(../jarvis_hx_output/tensorboard_0729/hx_cocoh5_072906) ...
Epoch 12/20
1/1 [==============================] - 42s 42s/step - loss: 1.9893 - rpn_class_loss: 3.1015e-05 - rpn_bbox_loss: 0.0531 - mrcnn_class_loss: 1.2489 - mrcnn_bbox_loss: 0.3533 - mrcnn_mask_loss: 0.3339 - val_loss: 2.5298 - val_rpn_class_loss: 0.0037 - val_rpn_bbox_loss: 0.6923 - val_mrcnn_class_loss: 1.0111 - val_mrcnn_bbox_loss: 0.3286 - val_mrcnn_mask_loss: 0.4941
Copying local_tb_dir(/tmp/model_dir/hx_cocoh5_072906) to tb_logdir(../jarvis_hx_output/tensorboard_0729/hx_cocoh5_072906) ...
Epoch 13/20
1/1 [==============================] - 42s 42s/step - loss: 1.9890 - rpn_class_loss: 3.1015e-05 - rpn_bbox_loss: 0.0531 - mrcnn_class_loss: 1.2486 - mrcnn_bbox_loss: 0.3533 - mrcnn_mask_loss: 0.3339 - val_loss: 2.4033 - val_rpn_class_loss: 0.2304 - val_rpn_bbox_loss: 0.8963 - val_mrcnn_class_loss: 0.4981 - val_mrcnn_bbox_loss: 0.2219 - val_mrcnn_mask_loss: 0.5565
Copying local_tb_dir(/tmp/model_dir/hx_cocoh5_072906) to tb_logdir(../jarvis_hx_output/tensorboard_0729/hx_cocoh5_072906) ...
Epoch 14/20
1/1 [==============================] - 43s 43s/step - loss: 2.0100 - rpn_class_loss: 3.1015e-05 - rpn_bbox_loss: 0.0531 - mrcnn_class_loss: 1.2696 - mrcnn_bbox_loss: 0.3533 - mrcnn_mask_loss: 0.3339 - val_loss: 2.3523 - val_rpn_class_loss: 0.0709 - val_rpn_bbox_loss: 0.6666 - val_mrcnn_class_loss: 0.6068 - val_mrcnn_bbox_loss: 0.6744 - val_mrcnn_mask_loss: 0.3335
Copying local_tb_dir(/tmp/model_dir/hx_cocoh5_072906) to tb_logdir(../jarvis_hx_output/tensorboard_0729/hx_cocoh5_072906) ...
Epoch 15/20
1/1 [==============================] - 42s 42s/step - loss: 1.9987 - rpn_class_loss: 3.1015e-05 - rpn_bbox_loss: 0.0531 - mrcnn_class_loss: 1.2583 - mrcnn_bbox_loss: 0.3533 - mrcnn_mask_loss: 0.3339 - val_loss: 3.6528 - val_rpn_class_loss: 0.0688 - val_rpn_bbox_loss: 1.1541 - val_mrcnn_class_loss: 0.8620 - val_mrcnn_bbox_loss: 0.8272 - val_mrcnn_mask_loss: 0.7406
Copying local_tb_dir(/tmp/model_dir/hx_cocoh5_072906) to tb_logdir(../jarvis_hx_output/tensorboard_0729/hx_cocoh5_072906) ...
Epoch 16/20
1/1 [==============================] - 42s 42s/step - loss: 1.9898 - rpn_class_loss: 3.1015e-05 - rpn_bbox_loss: 0.0531 - mrcnn_class_loss: 1.2493 - mrcnn_bbox_loss: 0.3533 - mrcnn_mask_loss: 0.3339 - val_loss: 3.3256 - val_rpn_class_loss: 4.8933e-05 - val_rpn_bbox_loss: 0.9305 - val_mrcnn_class_loss: 0.7511 - val_mrcnn_bbox_loss: 0.7859 - val_mrcnn_mask_loss: 0.8581
```

I hope that if the model are not changed, the val_losses are all the same, so that the validation result has no relation to the data variation.
Who have an idea about that?
